internal/command/junit/junit.go

```diff
@@ -299,3 +346,24 @@ func suiteFilesAsSortedList(files map[string]*moduletest.File) []*moduletest.Fil
 	}
 	return sortedFiles
 }
+
+func getDiagString(diags tfdiags.Diagnostics, sources map[string][]byte) string {
+	var diagsStr strings.Builder
+	for _, d := range diags {
+		diagsStr.WriteString(format.DiagnosticPlain(d, sources, 80))
+	}
+	return diagsStr.String()
+}
+
+func getWarnings(diags tfdiags.Diagnostics) tfdiags.Diagnostics {
+	// Collect warnings
+	warnDiags := tfdiags.Diagnostics{}
+	if diags.HasWarnings() {
+		for _, d := range diags {
+			if d.Severity() == tfdiags.Warning {
+				warnDiags = warnDiags.Append(d)
+			}
+		}
+	}
+	return warnDiags
+}
```

Comment:
Nitpick - AFAICT the code here would work just as well without `HasWarnings()` - it would return empty slice as nothing would be appended.

On a separate note I am somewhat surprised we don't have `Warnings()` method yet on `tfdiags.Diagnostics` - so I think it may be another candidate for logic to extract there and just call from here but feel free to leave this for another PR if you like.